### DISCUSSION TITLE
funktion nach Update des Hoymiles WR

### DISCUSSION THREAD

**Discussion by Gururom** [hoylabs/OpenDTU-OnBattery#1415] [General]
Hallo,
wollte fragen ob es schon jemanden gibt, der seinen Hoymiles auf die neue Firmware geupdatet hat, welche wohl bewirkt das am Ausgang geregelt wird und nicht am DC Eingang?
Also das beim runterregeln nicht mehr wichtig ist, dass alle Ports belegt sind, oder das auch stark unterschiedliche Leistung von den Modulen kommt und die Regelung trotzdem sauber funktioniert.

Tja eben ob das auch mit onBattery gut läuft, 
So könnte man einen an allen Ports belegten HMS16oo 4T auch mit einem Speicher an nur zwei Ports betreiben und Dieser würde trotzdem mit richtigen Werten, zBsp. Nachts geregelt werden können??

**Comment by Harisjems** [comment]
Es funktioniert wie gehabt hier.
![Screenshot_20241123_142126_Samsung Internet](https://github.com/user-attachments/assets/c04bb0d4-80c5-49c2-9f48-fb989e1f0116)

**Reply by Gururom** [reply]
Ich trau mich noch nicht, habe im Moment sogar Probleme mit der reglung mit der neusten onBattery version.
Ich wollte 2 WR regeln lassen, dass läuft nicht so.
Nin wieder zurück

**Comment by Harisjems** [comment]
Das Wetter ist im Moment nicht DPL-freundlich, meistens fahren hier alle 4 WR 100 %.
Ich teste gerade die Multiple WR-Regelung von hier:

https://github.com/hoylabs/OpenDTU-OnBattery/actions/runs/11881836875
![Screenshot_20241123_145753_Samsung Internet](https://github.com/user-attachments/assets/90790718-0526-41d1-8469-4d2806fb7eb2)
![Screenshot_20241123_145512_Samsung Internet](https://github.com/user-attachments/assets/e30824b7-6470-42d5-9145-99a38f3d5007)


Parallel zum OpenDTU-Client
Beide Varianten funktionieren.

**Comment by Settl3r** [comment]
Ich denke, es geht um die Sache mit der eingeführten Encryption. Auslesen der Daten soll wohl gehen, aber keine dynamische Änderung des WR-Limits z.B. Würde mich auch interessieren.

**Reply by stefan123t** [reply]
Die Encryption bezieht sich auf das Firmware Image auf dem Weg von der S-Miles Cloud zur Hoymiles DTU Lite/Pro S. Die Übertragung zwischen DTU und WR erfolgt nach wie vor unverschlüsselt. Sonst müsste der WR die Firmware entschlüsseln und validieren können.
Das erfolgt nach wie vor nur anhand der CRC16 Checksumme pro Paketbündel.

Anpassen des Limits funktioniert ebenso nach wie vor. Ergo es ist durch die Neuerung nur die Kommunikation zwischen S-Miles Cloud und Hoymiles DTU betroffen. Nicht die Kommunikation zwischen WR und OpenDTU.

Bitte keine unverifizierten Informationen streuen. Danke!

**Reply by Settl3r** [reply]
ok. Danke für den Hinweis. Dann war das in einigen anderen Quellen falsch beschrieben.

**Comment by Harisjems** [comment]
Ich würde es bei besserem Wetter beobachten. Im Moment ist DPL machtlos bei ca. 400 W Grundlast und 150 W vom Himmel bei 5 kWp.
Zwei Marstek B2500 Batterien sind nicht vorteilhaft bei 1,5 kW Tageseintrag. :-)

**Reply by Settl3r** [reply]
Ja, das wird erst wieder im Frühjahr interessant.

**Comment by Harisjems** [comment]
Dann werden die alle vier Eingänge auf HMS1600 über zwei Marstek-Speicher begrenzt. :-)
Ich will aber den Teufel nicht an die Wand klatschen oder malen. :-)

Edit:
Steuerung von DPL funktioniert weiter hin, gerade aus Neugierde getestet 


https://github.com/user-attachments/assets/3cc3ca36-8386-427f-ac0a-d730f518f8aa

**Reply by stefan123t** [reply]
Danke fürs Testen und die differenzierte Darstellung des DPL im Video.

Das bestätigen im Übrigen auch die anderweitig getraceden DTU Kommunikationsdaten (ua des ActivePowerLimit Kommandos) und die ersten Traces des Firmwareupgrades.

**Comment by ghost** [comment]
DPL funktioniert hier auf einem HMS-1600-4T auch ohne Probleme soweit.

![hms1600_dpl](https://github.com/user-attachments/assets/0d912e14-e0f4-413d-ae0b-df8c384f4097)

![hms1600_firmware](https://github.com/user-attachments/assets/1a9d8864-d826-4514-ac64-a790130b44d3)

**Comment by ghost** [comment]
Einzig, der Name des Netzprofils wird nicht (korrekt) angezeigt, seit der Umstellung auf "AT_TOR_Erzeuger_cosphi=1". Die Werte im Netzprofil sind aber soweit korrekt. .. hat jetzt natürlich nichts mit dem DPL zu tun.

![hms1600_grid_profile](https://github.com/user-attachments/assets/8bc9fe45-727c-4136-8f1b-f4c9ad6c76a5)

--- edit ---

Hier noch die Daten des Netzprofils, falls von Interesse:

`0c 03 20 00 00 0b 08 fc 07 30 00 0f 09 f9 00 01 02 3f 00 32 0a 55 00 08 09 f9 10 00 13 88 12 8e 00 01 14 1e 00 01 20 00 00 01 30 03 02 58 09 cb 07 a3 13 92 12 8e 40 00 07 d0 00 10 50 00 00 01 13 9c 01 90 00 32 60 04 00 01 09 e2 0a 10 08 7e 80 00 00 00 08 44 01 2c 08 a0 09 6f 09 b4 01 2c 90 00 00 01 00 64 b0 00 00 00 01 f4 00 5f 70 02 00 01 27 10 a0 02 00 00 00 00 8b 9d`

**Reply by Gururom** [reply]
na gut , dann werde ich mal meine org. DTU anschließen und auch ein Update anfragen
